UOP1_Project/Assets/Scripts/Tests/Play Mode/InputReader/KeyMappings.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Controls;
+
+namespace Tests.Play_Mode.InputReader
+{
+	public interface IMappingProvider
+	{
+		ButtonControl MoveLeft { get; }
+		ButtonControl MoveRight { get; }
+		ButtonControl MoveUp { get; }
+		ButtonControl MoveDown { get; }
+
+		ButtonControl Jump { get; }
+		ButtonControl Interact { get; }
+		ButtonControl Attack { get; }
+		ButtonControl Extra { get; }
+
+		ButtonControl Pause { get; }
+	}
+
+	public static class MappingProviderFactory
+	{
+		public static IMappingProvider Create(string type)
+		{
+			switch (type)
+			{
+				case nameof(KeyboardMappings):
+					return new KeyboardMappings();
+				case nameof(GamepadMappings):
+					return new GamepadMappings();
+				default:
+					throw new ArgumentException("string provided must be the nameof() some IMappingProvider. e.g. KeyboardMappings or GamepadMappings");
+			}
+		}
+	}
+
+	public class KeyboardMappings : IMappingProvider
+	{
+		public ButtonControl MoveLeft { get; private set; } = Keyboard.current.aKey;
+		public ButtonControl MoveRight { get; private set; } = Keyboard.current.aKey;
```

Comment:
Are left and right controls suppose to both be set to aKey? Wouldn't we expect Keyboard.current.dKey for MoveRight?